lib/ueberauth/strategies/helpers.ex

```diff
@@ -241,14 +242,18 @@ defmodule Ueberauth.Strategy.Helpers do
     |> List.first()
   end
 
-  defp get_forwarded_host_header(conn) do
-    conn
-    |> get_req_header("x-forwarded-host")
-    |> List.first()
+  defp get_host_header(conn) do
+    case get_req_header(conn, "x-forwarded-host") do
+      [] ->
+        get_req_header(conn, "host")
+        |> List.first()
+      [host | _] ->
+        host
+    end
   end
 
-  defp normalize_port("https", 80), do: 443
-  defp normalize_port(_, port) when is_integer(port), do: port
+  defp normalize_port("https", nil), do: 443
+  defp normalize_port(_, nil), do: 80
```

Comment:
`URI` provides a function for exactly this: https://hexdocs.pm/elixir/1.12/URI.html#default_port/1

Reply:
correct, i saw it just an hour ago and wanted to update the code 😆. Will add a test or two and update to use the URI function